frontend/alert/components/Timeline/index.tsx

```diff
@@ -0,0 +1,287 @@
+        const sameDate =
+            index == 0 ||
+            getMonthDay(courseData[index]["created_at"]) ==
+                getMonthDay(courseData[index - 1]["created_at"]);
+
+        if (item["old_status"] == "C" && item["new_status"] == "O") {
+            return ans.concat([[item, "opened", sameDate]]);
+        } else {
+            return ans.concat([[item, "closed", sameDate]]);
+        }
+    }, []);
+
+    return formattedData;
+};
+
+interface TimelineProps {
+    courseCode: string | null;
+    setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+const Timeline = ({ courseCode, setTimeline }: TimelineProps) => {
+    const [courseStatusData, setCourseStatusData] = useState([]);
+    const [loaded, setLoaded] = useState(false);
+    const [scrollTimeline, setScrollTimeline] = useState(false);
+
+    const close = !courseCode;
+
+    //hook that detects when a user is scrolling a component & when they stop scrolling
+    const useScroll = (ref) => {
+        useEffect(() => {
+            var timer;
+            const handleScroll = (e) => {
+                if (!scrollTimeline) {
+                    setScrollTimeline(true);
+                    clearTimeout(timer);
+                    timer = setTimeout(function () {
+                        setScrollTimeline(false);
+                    }, 1000);
+                }
+            };
+
+            ref.current.addEventListener("scroll", handleScroll, true);
+        }, [ref]);
+
+        return ref;
+    };
+
+    useEffect(() => {
+        if (!courseCode) {
+            return;
+        }
+
+        //loading course status update data
+        setLoaded(false);
+
+        fetch(`/api/base/statusupdate/${courseCode}/`).then((res) =>
+            res.json().then((courseResult) => {
+                //sort data by when it was created from earliest to latest
+                courseResult.sort((a, b) =>
+                    a.created_at < b.created_at ? 1 : -1
+                );
+
+                setCourseStatusData(formatData(courseResult));
+
+                setLoaded(true);
+            })
+        );
+    }, [courseCode]);
+
+    return (
+        <AlertHistoryContainer
+            close={close}
+            ref={useScroll(
+                useOnClickOutside(
+                    () => {
+                        setTimeline(null);
+                        courseCode = null;
+                    },
+                    close,
+                    "historyIcon"
+                )
+            )}
+        >
+            <AlertTitle>Alert History</AlertTitle>
+            <CloseButton
+                onClick={() => {
+                    setTimeline(null);
+                    courseCode = null;
+                }}
+            >
+                <FontAwesomeIcon icon={faTimes} />
+            </CloseButton>
+
+            {/* Only show if loaded */}
+            {loaded && courseStatusData && courseStatusData.length > 0 ? (
```

Comment:
this is a pretty long conditional, would think about pulling these out into their own component definitions. Not necessarily their own files, but just pull them above for readability.